server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -89,6 +92,15 @@
 
     private static final TimeValue WAIT_FOR_COMPLETION_POLL = timeValueMillis(100);
 
+    private static final String TASK_RESOURCE_CONSUMERS_ATTRIBUTES = "cluster.task.consumers";
+
+    private static final Setting<Boolean> TASK_RESOURCE_CONSUMERS_SETTING = Setting.boolSetting(
+        TASK_RESOURCE_CONSUMERS_ATTRIBUTES,
+        true,
```

Comment:
should the default be `false` here ? We should do some performance test before enabling by default ?

Reply:
Enabled it by default. Added performance results to this PR.